api/v1alpha1/ext_auth_types.go

```diff
@@ -0,0 +1,129 @@
+	// Host is the hostname of the gRPC External Authorization service.
+	Host gwapiv1a2.PreciseHostname `json:"host"`
+
+	// Port is the network port of the gRPC External Authorization service.
+	Port gwapiv1a2.PortNumber `json:"port"`
+
+	// TLS defines the TLS configuration for the gRPC External Authorization service.
+	// Note: If not specified, the proxy will talk to the gRPC External Authorization
+	// service in plaintext.
+	// +optional
+	TLS *TLSConfig `json:"tls,omitempty"`
+}
+
+// HTTPExtAuthService defines the HTTP External Authorization service
+type HTTPExtAuthService struct {
+	// Host is the hostname of the HTTP External Authorization service.
+	Host gwapiv1a2.PreciseHostname `json:"host"`
+
+	// Port is the network port of the HTTP External Authorization service.
+	// If port is not specified, 80 for http and 443 for https are assumed.
+	Port *gwapiv1a2.PortNumber `json:"port,omitempty"`
+
+	// Path is the path of the HTTP External Authorization service.
+	// If path is specified, the authorization request will be sent to that path,
+	// or else the authorization request will be sent to the root path.
+	Path *string `json:"path,omitempty"`
+
+	// TLS defines the TLS configuration for the HTTP External Authorization service.
+	// Note: If not specified, the proxy will talk to the HTTP External Authorization
+	// service in plaintext.
+	// +optional
+	TLS *TLSConfig `json:"tls,omitempty"`
+
+	// HeadersToBackend are the authorization response headers that will be added
+	// to the original client request before sending it to the backend server.
+	// Note that coexisting headers will be overridden.
+	// If not specified, no authorization response headers will be added to the
+	// original client request.
+	// +optional
+	HeadersToBackend []string `json:"headersToBackend,omitempty"`
+}
+
+// TLSConfig describes a TLS configuration.
+type TLSConfig struct {
+	// CertificateRef is the reference to a Kubernetes Secret that contains the
+	// TLS certificate and private key. The certificate and private key will be
+	// used to establish a TLS handshake between the Envoy proxy and the external
+	// authorization server.
+	// The referenced Secret must contain two keys: tls.crt and tls.key.
+	//
+	// If this field is not specified, the Envoy proxy will not present a client
+	// certificate to the external authorization server.
+	//
+	// +optional
+	CertificateRef *gwapiv1a2.SecretObjectReference `json:"certificateRef,omitempty"`
+
+	// CACertRef is the reference to a Kubernetes ConfigMap that contains a
+	// PEM-encoded TLS CA certificate bundle, which is used to validate the
+	// certificate presented by the external authorization server.
+	// The referenced ConfigMap must contain a key named ca.crt.
+	//
+	// If not specified, the proxy will use the system default certificate pool to
+	// verify the server certificate.
+	//
+	// +optional
+	CACertRef *gwapiv1.LocalObjectReference `json:"caCertRef,omitempty"`
```

Comment:
```suggestion
	CACertificateRefs *gwapiv1.LocalObjectReference `json:"caCertRef,omitempty"`
```

Reply:
lets keep this identical to existing API field https://github.com/envoyproxy/gateway/blob/236cba58b587e195728e779f06cc85e249ff43d2/api/v1alpha1/tls_types.go#L127C2-L127C81

Reply:
After discussing with @arkodg, we decided to remove all the fields in the TLSConfig for now and address them later in a separate issue since we need to confirm with Gateway API upstream how the client certificate should be specified for Backends.